enclave/core/switchlesscalls.c

```diff
@@ -44,6 +44,9 @@ oe_result_t oe_handle_init_switchless(uint64_t arg_in)
     oe_switchless_call_manager_t safe_manager;
     size_t contexts_size, threads_size;
 
+    if (oe_is_switchless_initialized())
```

Comment:
Is this thread-safe?

Reply:
cc @jxyang Why is oe_handle_init_switchless exposed in a public header? From the host `oe_start_switchless_manager()` is only in an internal header.

And then from a philosophy perspective @anakrish, if a function is internal and we only call it once, can we safely ignore thread safety even though someone could technically include an internal header and call it?

Reply:
@jhand2 The reason for making it thread-safe is from a security perspective. Our philosophy is that enclaves don't trust the host; a host can intentionally trigger code paths that we have not anticipated.

Eventhough, oe_handle_init_switchless is an internal function, a host can launch two threads and make the OE_ECALL_INIT_CONTEXT_SWITCHLESS ecall. If we don't proactively guard against such malicious calls, the behavior of the enclave is undefined, potentially causing confidentiality vulnerabilities.

Reply:
That makes sense. I'll add proper logging :smile: